File: t-tasks/src/main/java/com/teo/ttasks/services/MyGcmJobService.java
package com.teo.ttasks.services;

import android.support.annotation.NonNull;

import com.birbit.android.jobqueue.JobManager;
import com.birbit.android.jobqueue.scheduling.GcmJobSchedulerService;

import javax.inject.Inject;

import dagger.android.AndroidInjection;

public class MyGcmJobService extends GcmJobSchedulerService {

    @Inject JobManager jobManager;

    @Override @NonNull
    protected JobManager getJobManager() {
        if (jobManager == null)
            AndroidInjection.inject(this);
        return jobManager;
    }
}

File: t-tasks/src/main/java/com/teo/ttasks/services/MyGcmJobService.kt
package com.teo.ttasks.services

import com.birbit.android.jobqueue.JobManager
import com.birbit.android.jobqueue.scheduling.GcmJobSchedulerService
import dagger.android.AndroidInjection
import javax.inject.Inject

class MyGcmJobService : GcmJobSchedulerService() {

    @Inject internal lateinit var jobManager: JobManager

    override fun getJobManager(): JobManager {
        AndroidInjection.inject(this)
        return jobManager
    }
}
